### PR TITLE
#161656449 Fix room selection on request modal

### PR DESCRIPTION
**Pull request by Stylll**
#### What does this PR do?
Fixes issues with request modal and room selection

#### Description of Task to be completed?
- Prevent user from selecting room currently occupied by the opposite gender
- Prevent available rooms list in each trip detail row from affecting one another (changes made in one trip detail row updates the other existing rows)
- Enable updating request when the 'No Beds Found' option is selected

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_front.git`
- Enter into the project directory - `cd travel_tool_front`
- checkout the branch `git checkout bg-change-guest-room-161744068`
- Start the server with `make start`
- Start the backend of this project using this link [Click here](https://github.com/andela/travel_tool_back/pull/169)
- Visit https://travela-local.andela.com
- Login to the application
- Navigate to the request page
- Test out the fixes listed in the description above

#### What are the relevant pivotal tracker stories?
[#161656449](https://www.pivotaltracker.com/story/show/161656449)

#### Questions:

### CODE REVIEW

File: src/components/Forms/NewRequestForm/FormFieldsets/__tests__/TravelDetailsItem.test.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+const setup = (props) => shallow(<TravelDetailsItem {...props} />);
+
+describe('Test Suite for <TravelDetailsItem />', () => {
+  it('should match snapshot', () => {
+    const wrapper = setup(props);
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  // it('should match snapshot when selection is multi', () => {
```
Comment (loicemeyo) [inline src/components/Forms/NewRequestForm/FormFieldsets/__tests__/TravelDetailsItem.test.js]: @Stylll should this be taken off or updated?

**Comment by Stylll** [comment]
@asamisellah No it does not. The user will have to click on the available beds listbox to select a new bed. Also, the backend is being worked on to validate cases like this.
Furthermore, a PR is being worked on to automatically fetch new beds once any of the required fields are changed.

File: src/components/Forms/NewRequestForm/FormFieldsets/__tests__/TravelDetailsItem.test.js (hunk 1 above)
Comment (Stylll) [reply src/components/Forms/NewRequestForm/FormFieldsets/__tests__/TravelDetailsItem.test.js]: @loicemeyo Thanks for pointing this out. I have removed the test case.

**Comment by amarachukwu-agbo** [comment]
@samuelkubai , LGTM

**Comment by samuelkubai** [comment]
@Stylll please resolve conflicts

**Comment by Stylll** [comment]
@samuelkubai the merge conflicts have been fixed.

**Pull request merged**